L1Trigger/Phase2L1ParticleFlow/plugins/L1TCorrelatorLayer1Producer.cc

```diff
@@ -347,19 +357,32 @@ void L1TCorrelatorLayer1Producer::produce(edm::Event &iEvent, const edm::EventSe
     edm::Handle<std::vector<l1t::Vertex>> vtxHandle;
     iEvent.getByToken(extTkVtx_, vtxHandle);
     for (const auto &vtx : *vtxHandle) {
+      ptsums.push_back(std::pair<float, float>(vtx.pt(), vtx.z0()));
       if (ptsum == 0 || vtx.pt() > ptsum) {
         ptsum = vtx.pt();
         z0 = vtx.z0();
       }
     }
     pvwd = l1t::VertexWord(1, z0, 1, ptsum, 1, 1, 1);
   }
-
   l1ct::PVObjEmu hwpv;
   hwpv.hwZ0 = l1ct::Scales::makeZ0(pvwd.z0());
   event_.pvs.push_back(hwpv);
   event_.pvs_emu.push_back(pvwd.vertexWord());
-
+  //Do a quick histogram vertexing to get multiple vertices (Hack for the taus)
+  if (NVtx_ > 1) {
+    std::stable_sort(ptsums.begin(), ptsums.end(), [](const auto &a, const auto &b) { return a.first > b.first; });
+    for (int i0 = 0; i0 < std::min(int(ptsums.size()), int(NVtx_)); i0++) {
+      z0s.push_back(ptsums[i0].second);
+    }
+    if (addExtraVtx_)
+      doVertexings(z0s);
+    for (unsigned int i = 1; i < z0s.size(); ++i) {
```

Comment:
If addExtraVtx_ is true, then you're no longer guaranteed that the first vertex is the one already added in `event_.pvs.push_back(hwpv);` above, so by skipping the first of the z0s you may be skipping one vertex and duplicating another one.
So, I think you need

     if (addExtraVtx_) {
         event_.pvs.clear();
         doVertexings(z0s);
     }
    for (unsigned int i = (addExtraVtx_ ? 0 : 1); i < z0s.size(); ++i) {

That said, if you get ok performance of the 2 vertex case by just changing the configuration in `L1Trigger/L1TTrackMatch/python/L1TkPrimaryVertexProducer_cfi.py` then I would prefer to remove this entirely

Reply:
Agreed, this was really a hacky fix when the vertexer was only outputting 1 vertex. Updated.